ssl/statem/extensions_srvr.c

```diff
@@ -806,7 +806,8 @@ int tls_parse_ctos_psk(SSL *s, PACKET *pkt, unsigned int context, X509 *x,
 int tls_construct_stoc_renegotiate(SSL *s, WPACKET *pkt, unsigned int context,
                                    X509 *x, size_t chainidx, int *al)
 {
-    if (!s->s3->send_connection_binding)
+    if (!s->s3->send_connection_binding ||
+        (s->options & SSL_OP_NO_RENEGOTIATION))
```

Comment:
Hmm. I'm wondering whether this is the right thing to do. Perhaps some clients might abort if they don't negotiate secure reneg (even if they never then use it)?